### ISSUE TITLE
[HOLD for payment 2023-04-26] [$1000] Android - Double click to close phone popup reopens the popup

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#16582]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Open any report
3. Click on phone icon
4. Double click anywhere besides the link in popup

## Expected Result:
App should close the phone popup

## Actual Result:
App closes the popup and after few seconds it reopens the popup

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.90.4

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/228044863-1676029f-aaef-4a74-94b5-741b83463166.mp4

https://user-images.githubusercontent.com/93399543/228044877-f69dd5e3-59e3-49bf-b9e3-ec648e9e8784.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:**@dhanashree-sawant

**Slack conversation:**https://expensify.slack.com/archives/C049HHMV9SM/p1679905062814329

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019aa5b3ba0f811864</li>
        <li>Upwork Job ID: 1643904955903094784</li>
        <li>Last Price Increase: 2023-04-13</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @johncschuster (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@johncschuster, @Julesssss Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@johncschuster, @Julesssss Still overdue 6 days?! Let's take care of this!

**Comment by Julesssss** [comment]
Hey @johncschuster. I was able to reproduce this, would you mind completing the final bug0 triage list items so we can open it up for proposals? Thanks!

**Comment by johncschuster** [comment]
Done! Let's open it up!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~019aa5b3ba0f811864

**Comment by MelvinBot** [comment]
Current assignee @johncschuster is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @aimane-chnaif (`External`)

**Comment by MelvinBot** [comment]
Current assignee @Julesssss is eligible for the External assigner, not assigning anyone new.

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Double-click(press) reopens the VideoChatMenu popover when we are trying to close it on smaller screens.

### What is the root cause of that problem?
Currently, we are toggling the menu popover on the outside click for onClose event too and 2 clicks(on smaller screens due to animation time) will toggle 2 times which again makes the popover to be visible.
https://github.com/Expensify/App/blob/55d60985385319ed9e1b06a0eb3669121cbbf3c7/src/components/VideoChatButtonAndMenu/BaseVideoChatButtonAndMenu.js#L128-L129

### What changes do you think we should make in order to solve the problem?
Instead of toggling for the onClose we can add a new function called `hideVideoChatMenu` which sets the state prop `isVideoChatMenuActive` to false and bind it to the `Popover` `onClose` prop.
```
    hideVideoChatMenu() {
        this.setState({isVideoChatMenuActive: false});
    }
```
```
     onClose={this.hideVideoChatMenu}
```

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When double tapping outside of the video chat popup, app closes the popup and after few seconds it reopens the popup

The same issue can be seen if we tap on the `video chat button` twice, it will open and then close.

### What is the root cause of that problem?
The problem is in this line https://github.com/Expensify/App/blob/6331bd00a4505432875ee7d4bad02a0fb0266c85/src/components/VideoChatButtonAndMenu.js#L90 where we're toggling the popup when clicking outside the Popover.

When double tapping, the `toggle` is triggered twice, leading to the popup closes and then (toggled to) open a few seconds later.

Same for when clicking the `video chat button` twice.

### What changes do you think we should make in order to solve the problem?


We need to explicitly pass in the boolean visibility state of the video chat menu when calling `toggle`. This is what we're doing for all other `toggle` style function in other components.

The `toggleVideoChatMenu` should accept a new `value` boolean param and will set the `isVideoChatMenuActive` state according to that param.

All places that use `toggleVideoChatMenu` need to pass in true/false based on whether the video chat menu should be visible after the action.

Specifically, passing `false` in
https://github.com/Expensify/App/blob/78a9b540f6244276599d58f9aa3fc7d671441c8d/src/components/VideoChatButtonAndMenu/BaseVideoChatButtonAndMenu.js#L129
https://github.com/Expensify/App/blob/78a9b540f6244276599d58f9aa3fc7d671441c8d/src/components/VideoChatButtonAndMenu/BaseVideoChatButtonAndMenu.js#L45
https://github.com/Expensify/App/blob/78a9b540f6244276599d58f9aa3fc7d671441c8d/src/components/VideoChatButtonAndMenu/BaseVideoChatButtonAndMenu.js#L53

and `true` in 
https://github.com/Expensify/App/blob/78a9b540f6244276599d58f9aa3fc7d671441c8d/src/components/VideoChatButtonAndMenu/BaseVideoChatButtonAndMenu.js#L117

### What alternative solutions did you explore? (Optional)
NA

**Comment by MelvinBot** [comment]
@johncschuster @Julesssss @aimane-chnaif this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
@johncschuster @Julesssss @aimane-chnaif this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by Julesssss** [comment]
> I'd like to clarify the scope here. Should we fix only backdrop case or menu items as well?

@aimane-chnaif good point. I'd say we should aim to solve this in as many places as possible.

**Comment by Pujan92** [comment]
Yes @aimane-chnaif , I think your mentioned case is of multiple finger whereas this issue I assume about the same finger double click.

**Comment by Pujan92** [comment]
> Both cases should be handled here. Btw, your proposal still doesn't cover the case of same finger double click.

In multiple fingers click the popup reopening issue doesn't persist, it just blur the focus from the first click of the menuItem(IMO that is not an issue).
My proposal covers the double-click popover reopen issue.

https://user-images.githubusercontent.com/14358475/231137197-edd8007d-c9d4-4f42-b61f-ad5496a3184c.mp4

**Comment by tienifr** [comment]
@aimane-chnaif Just checking what do you think of the approach to make the `toggleVideoChatMenu` idempotent [here](https://github.com/Expensify/App/issues/16582#issuecomment-1498762867). We're doing that for other similar features.

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by MelvinBot** [comment]
@johncschuster, @Julesssss, @aimane-chnaif Huh... This is 4 days overdue. Who can take care of this?

**Comment by Julesssss** [comment]
> yes, that will do the job but I am checking possibility to fix this in root level.

Just bumping this @aimane-chnaif.

**Comment by Julesssss** [comment]
OKay, lets stick with the simpler solution that we're already using elsewhere. Thanks all for the proposals. 

Picking @Pujan92 in this case because both of the state solutions are pretty similar.

**Comment by MelvinBot** [comment]
📣 @Pujan92 You have been assigned to this job by @Julesssss!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by tienifr** [comment]
noted, if we think the cases I mentioned are out of scope for this GH issue, I’ll report them as a separate bug.

Thanks!

**Comment by tienifr** [comment]
ok so it seems in [the PR](https://github.com/Expensify/App/pull/17498) we decided to fix all the cases that I mentioned in [my proposal](https://github.com/Expensify/App/issues/16582#issuecomment-1498762867) and also pretty much [used the exact implementation](https://github.com/Expensify/App/pull/17498#discussion_r1168679780) that I proposed.

@aimane-chnaif @Julesssss do you think it'll be more fair if we split the compensation here? I'd be happy with it since @Pujan92 already created the PR.

I'd be good if my proposal was not accepted because the cases it covered are out of the scope of this GH issue. But we ended up _fixing all_ those cases as well so they _are_ in the scope of the GH issue.

**Comment by tienifr** [comment]
Also FYI all the precedences in other issues that I've seen so far favored a complete solution that fixes the full GH issue scope, in case the fix is straight forward, for example [here](https://github.com/Expensify/App/issues/15924#issuecomment-1470551318) and [here](https://github.com/Expensify/App/issues/16711#issuecomment-1493223047).

**Comment by Julesssss** [comment]
> we decided to fix all the cases that I mentioned in https://github.com/Expensify/App/issues/16582#issuecomment-1498762867 and also pretty much https://github.com/Expensify/App/pull/17498#discussion_r1168679780 that I proposed.

I think either choice would be seen as unfair to someone. The extra details in your proposal were good, but IMO both proposals were suggesting the same solution. So a 50/50 split doesn't seem that fair to me, given that your proposal followed an existing one and you didn't have to complete the checklist.

**Comment by tienifr** [comment]
@Julesssss I’m not seeing them as same solutions. The selected proposal clearly only works for 1 case out of 3 cases that are finally addressed in the PR, also the implementations are different. 

What I saw was, we usually favored solution that’s well-thought-of and helps to fix the issue in the broadest scope possible, which will likely create a better outcome for the product.

With that said, I’ll happily accept any decision made here.

Thanks

**Comment by Julesssss** [comment]
Yeah, I do see your point. I didn't notice that the PR had the exact diff.

Typically I might go with a more detailed proposal, but I don't want to encourage contributors to 'swipe' existing proposals just by adding more details, or to add the full diff to the prosal when it isn't necessary. **I don't think that's what happened here**, but it's just why I try not to select similar proposals that are added after someone has already identified the root cause. 

In this case, I could have gone either way due to the simplicity of the cause/fix and unfortunately, it went against you this time. But let's see what @Pujan92 thinks.

**Comment by johncschuster** [comment]
@aimane-chnaif and @Pujan92, can you apply to the Upwork job when you get a chance?

https://www.upwork.com/jobs/~019aa5b3ba0f811864

**Comment by dhanashree-sawant** [comment]
Hi @johncschuster, should I also apply for reporting bonus?

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.1-3 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17498


If no regressions arise, payment will be issued on 2023-04-26. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@aimane-chnaif] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@aimane-chnaif] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@aimane-chnaif] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@johncschuster] Determine if we should create a regression test for this bug.
- [ ] [@aimane-chnaif] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@johncschuster] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by dhanashree-sawant** [comment]
Hi @johncschuster, pinging for upwork (Also let me know if I should also apply to the job for reporting bonus).

**Comment by johncschuster** [comment]
Hi @dhanashree-sawant! Yes, please!

**Comment by johncschuster** [comment]
@aimane-chnaif and @Pujan92, I've sent offers to you both. Can you please accept them?

@dhanashree-sawant, once you apply, I'll do the same for you as well!

**Comment by dhanashree-sawant** [comment]
Thanks @johncschuster, I have submitted the proposal

**Comment by johncschuster** [comment]
Thanks, @dhanashree-sawant! I've sent the offer!

**Comment by johncschuster** [comment]
@aimane-chnaif, are you able to get to your portion of the [BugZero checklist](https://github.com/Expensify/App/issues/16582#issuecomment-1515077101) above?

**Comment by johncschuster** [comment]
@aimane-chnaif bump

**Comment by johncschuster** [comment]
Thanks, @aimane-chnaif!

**Comment by johncschuster** [comment]
@dhanashree-sawant, @aimane-chnaif, @Pujan92 

Payment has been issued! Thank you!

**Comment by johncschuster** [comment]
@Julesssss, I think we're good to close this, right?
